### PR TITLE
Modules Updates

### LINKED ISSUE #7696: Fax/SMS Weno modules needs updates and bug fixes.
### Describe the bug
Module has some bugs like 
- batch notifications fails
- need to add option to add a background task for auto notifications.
- fix for allowing appointment status update for recurring events in batch notifications.
- Add menu item to use new email modules batch notification option instead of running from module setup.
- Add some debugging options to Weno.
- Refactor Weno Download Log to give better status of various tasks.
- Modify Prescription sync to review last sync date and ensure to never miss a sync due to internet or Weno server issues.

### Client configuration

- Browser:
- OpenEMR version: 7.0.2
- Operating system: all

### PR DESCRIPTION
**Pull request by sjpadgett**
include RC SDK in module. Prevents running composer. 
refactor weno syncs for UTC dates
add weno download logs and import csvs
allow weno users to select location in erx compose. refactor faxsms menus under a top level menu Services. 
fixed batch appointment notifications to include email. add a send and test email appointment reminders under new Services for modules menu 
add sqlUpgrade function in the listener service class for other to copy if needed. reviewed and changed where needed ACLs

Fixes #7696

### CODE REVIEW

**Comment by sjpadgett** [comment]
I really didn't have a choice but to add the RC SDK in module.
Several user could not install module because they either couldn't or wouldn't install and run composer.
Lots of files but only requires about 1.5MB compressed to patch.

**Comment by sjpadgett** [comment]
@bradymiller This isn't going to work is it? I may just go ahead and pull the module and offer as a download!

File: interface/modules/custom_modules/oe-module-weno/templates/download_log_viewer.php [hunk 1]
```diff
@@ -117,11 +120,21 @@ function downloadPresLog(){
                 }
             });
         }
+
+        function areYouSure() {
+            let yn = confirm('<?php echo xlt("Are you sure you want to download logs?"); ?>');
```
Comment (bradymiller) [inline interface/modules/custom_modules/oe-module-weno/templates/download_log_viewer.php:125]: xlj (and drop outside quotes)

File: interface/modules/custom_modules/oe-module-weno/templates/weno_fragment.php [hunk 2]
```diff
@@ -63,41 +61,95 @@
...
-        <a class="mr-2" onclick="top.restoreSession();" href="<?php echo $GLOBALS['webroot'] ?>/interface/modules/custom_modules/oe-module-weno/templates/indexrx.php"><span><i class="fa fa fa-pencil-alt mr-1"></i><?php echo xlt("Prescribe"); ?></span></a>
-    </div>
-</div>
+<script>
+    function setPrescribeLocation() {
+        const facilitySelect = document.getElementById('facilitySelect');
+        let newLocation = facilitySelect.value;
+
+        if (!newLocation) {
+            alert("Please select a facility before prescribing.");
```
Comment (bradymiller) [inline interface/modules/custom_modules/oe-module-weno/templates/weno_fragment.php:119]: translate

File: interface/modules/custom_modules/oe-module-weno/templates/weno_fragment.php [hunk 3]
```diff
@@ -63,41 +61,95 @@
...
+    </span>
+</div>
+<br />
+<?php
+if ($reSync === true) {
+    // Trigger the sync_report function to sync the patient's prescriptions
+    $reSync = '';
+    $_GET['resync'] = '';
+    unset($_GET['resync']);
+    echo "<script>sync_report($pid)</script>";
```
Comment (bradymiller) [inline interface/modules/custom_modules/oe-module-weno/templates/weno_fragment.php:149]: text() around $pid

Comment (bradymiller) [reply interface/modules/custom_modules/oe-module-weno/templates/weno_fragment.php:149]: actually, this is javascript context, so would use js_escape() aroujnd $pid

File: interface/modules/custom_modules/oe-module-weno/templates/weno_fragment.php [hunk 4]
```diff
@@ -162,17 +215,32 @@ function getProviderByWenoId($external_id, $provider_id = ''): string
                 <?php } ?>
             </select>
         </div>
+        <div class="form-group">
+            <label role="button" id="label-location" class="text-primary mb-1 mr-1" for="facilitySelect" title="<?php echo $titleLocation ?>" data-toggle="popover" data-content="<?php echo $popoverLocation ?>">
```
Comment (bradymiller) [inline interface/modules/custom_modules/oe-module-weno/templates/weno_fragment.php:219]: I couldn't figure out where $titleLocation is. Just need to ensure it is escaped via attr()

**Comment by sjpadgett** [comment]
These are already escaped

**Comment by bradymiller** [comment]
In this case, would it make more sense to bring in ringcentral into the main core composer.json? So then won't have bunch of redundant third party packages in each vendor directory.

**Comment by sjpadgett** [comment]
I agree, Brady, but people do not want to run composer on a patch. It’s
starting to move more and more away from average user it don’t have
technology experience where you have to be savvy on the command line to
install opening MR.
However, you are correct and I have a plan, but for this patch, I need it
ring central they already in the module. So I will take care of it before
next release. And master it is being brought in via composer so that I can
move for module eventually.

**Comment by sjpadgett** [comment]
@bradymiller Sorry my text to speech on phone mangled message. Anyway I have a plan to remove lib. in next release. Also, sdk is in composer.json in patch.

**Comment by bradymiller** [comment]
hi @sjpadgett , Agree that need to avoid having to run composer for a patch. Sounds like a good plan :)

**Pull request merged**